### PR TITLE
First Kotlin Quickstart

### LINKED ISSUE #633: [Internal] Translate inappmessaging app to kotlin
See #557

### PR DESCRIPTION
**Pull request by samtstern**
See #557

Fixes #633 

@the-dagger
@rosariopfernandes

Here's how it looks:
chooser

### CODE REVIEW

File: inappmessaging/app/src/main/java/com/google/firebase/fiamquickstart/EntryChoiceActivity.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+import android.content.Intent;
+
+import com.firebase.example.internal.BaseEntryChoiceActivity;
+import com.firebase.example.internal.Choice;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class EntryChoiceActivity extends BaseEntryChoiceActivity {
```
Comment (the-dagger) [inline inappmessaging/app/src/main/java/com/google/firebase/fiamquickstart/EntryChoiceActivity.java]: unrelated : can we have this activity in Kotlin for the next quickstarters?

File: inappmessaging/app/src/main/java/com/google/firebase/fiamquickstart/KotlinMainActivity.kt [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+
+        firebaseAnalytics = FirebaseAnalytics.getInstance(this)
+        firebaseIam = FirebaseInAppMessaging.getInstance()
+
+        firebaseIam.isAutomaticDataCollectionEnabled = true
+        firebaseIam.setMessagesSuppressed(false)
+
+        findViewById<View>(R.id.event_trigger_button)
```
Comment (the-dagger) [inline inappmessaging/app/src/main/java/com/google/firebase/fiamquickstart/KotlinMainActivity.kt]: Using synthetic binding here might be a good idea, what do you think?

Comment (samtstern) [reply inappmessaging/app/src/main/java/com/google/firebase/fiamquickstart/KotlinMainActivity.kt]: Just learned about a new thing!  Thanks for showing me that.

**Comment by rosariopfernandes** [comment]
Looks better than what I imagined! :clap:

**Comment by samtstern** [comment]
@rosariopfernandes yes we definitely can.  Let's do the simplest ones first and work our way up.

Also so we don't step on each other's toes let's do the following:

  * I'll file an issue for each quickstart
  * Before starting a quickstart translation, comment in the issue that you're going to handle it
  * PRs should only be one quickstart at a time

Thank you for always offering your help!

**Comment by samtstern** [comment]
@rosariopfernandes 

https://github.com/firebase/quickstart-android/milestone/1

**Pull request merged**